pkg/kubeinteraction/secrets.go

```diff
@@ -99,13 +93,13 @@ func (k Interaction) GetSecret(ctx context.Context, secretopt GetSecretOpt) (str
 	return string(secret.Data[secretopt.Key]), nil
 }
 
-func getBasicAuthSecretName(runEvent *info.Event) string {
-	return fmt.Sprintf(basicAuthSecretName, strings.ToLower(runEvent.Organization), strings.ToLower(runEvent.Repository))
+func GetBasicAuthSecretName(runEvent *info.Event) string {
+	return strings.ToLower(fmt.Sprintf(basicAuthSecretName, strings.ToLower(runEvent.Organization), (runEvent.Repository), randomAlphaString(4)))
 }
 
 // DeleteBasicAuthSecret deletes the secret created for git-clone basic-auth
 func (k Interaction) DeleteBasicAuthSecret(ctx context.Context, logger *zap.SugaredLogger, runEvent *info.Event, targetNamespace string) error {
-	secretName := getBasicAuthSecretName(runEvent)
+	secretName := GetBasicAuthSecretName(runEvent)
```

Comment:
here we are calling `GetBasicAuthSecretName` won't it return a new secret name?

Reply:
ah yeah, you are right lol will fix thi in runEvent